File: careerforge-frontend/careerforge-frontend/src/services/studyLogService.js
import axiosConfig from "../api/axiosConfig";

const studyLogService = {

    getAll: () => axiosConfig.get("/studylogs"),

    create: (data) => axiosConfig.post("/studylogs", data),

    update: (id, data) => axiosConfig.put(`/studylogs/${id}`, data),

    delete: (id) => axiosConfig.delete(`/studylogs/${id}`)

};

export default studyLogService;

File: careerforge-backend/src/main/java/com/careerforge/service/StudyLogService.java
package com.careerforge.service;

import com.careerforge.entity.StudyLog;

import java.util.List;

public interface StudyLogService {

    StudyLog add(StudyLog log);

    List<StudyLog> getAll();

    StudyLog getById(Long id);

    StudyLog update(Long id, StudyLog log);

    void delete(Long id);

}